Make `value` for custom instruments easier to use (#5472)
apollographql/router: docs/source/configuration/telemetry/instrumentation/instruments.mdx

          type: counter
          unit: count
          description: "my description"

      graphql:
        acme.graphql.list.lengths:
          value:
            list_length: value
          type: histogram
          unit: count

>           description: "my description"

Reviewer: Line 82 should be updated to describe what this graphql instrument does/how it compares to the other three custom instruments.

Author: [7669dd2](https://github.com/apollographql/router/pull/5472/commits/7669dd2b783acacfa3addf7c9664270f134138ce)

#### `value`

> The `value` of an instrument is the value which will be drawn from and is dependent on the stage that the instrument is defined at.
> This can be one of the following:

Reviewer: The current definition of `value` is confusing and I'm not sure it adds anything. I think just the examples are enough to explain what it is.

```suggestion
An instrument's `value` is dependent on lifecycle stage that the instrument is defined at. `value` can be one of the following:
```

* `duration` - the duration of the pipeline service (router/supergraph/subgraph).
* `unit` - the number of times the pipeline service has been executed (router/supergraph/subgraph/graphql).
* `custom` - a custom value extracted from the pipeline service. See [selectors](./selectors) for more information. (router/supergraph/subgraph/graphql)
* `event_duration` - the duration of an event in the pipeline service (supergraph).
* `event_unit` - the number of times an event in the pipeline service has been executed (supergraph).

> * `event_custom` - a custom value extracted from the event in the pipeline service. See [selectors](./selectors) for more information (supergraph).

Reviewer: Assuming the parens denote which lifecycle stage the `value` is available at, this would be cleaner as a table (value / description / lifecycle stage availability).

LMK if you'd prefer for me to reformat it and I can push a commit.

Author: Yes, go ahead with any reformatting.

Reviewer: [51467a1](https://github.com/apollographql/router/pull/5472/commits/51467a162ea3cae08fa7433f8390b108429655d8)
